src/njs_cnst.c

```diff
@@ -0,0 +1,621 @@
+        njs_string_kw("toString", 0, 0),
+        njs_string_kw("toTimeString", 0, 0),
+        njs_string_kw("toUpperCase", 0, 0),
+        njs_string_kw("toUTCString", 0, 0),
+        njs_string_kw("trimEnd", 0, 0),
+        njs_string_kw("trimStart", 0, 0),
+        njs_string_kw("trim", 0, 0),
+        njs_string_kw("trunc", 0, 0),
+        njs_string_kw("type", 0, 0),
+        njs_string_kw("usec", 0, 0),
+        njs_string_kw("unscopables", 0, 0),
+        njs_string_kw("unshift", 0, 0),
+        njs_string_kw("utf-8", 0, 0),
+        njs_string_kw("value", 0, 0),
+        njs_string_kw("valueOf", 0, 0),
+        njs_string_kw("values", 0, 0),
+        njs_string_kw("version", 0, 0),
+        njs_string_kw("version_number", 0, 0),
+        njs_string_kw("write", 0, 0),
+        njs_string_kw("writable", 0, 0),
+        njs_string_kw("writeDoubleBE", 0, 0),
+        njs_string_kw("writeDoubleLE", 0, 0),
+        njs_string_kw("writeFloatBE", 0, 0),
+        njs_string_kw("writeFloatLE", 0, 0),
+        njs_string_kw("writeIntBE", 0, 0),
+        njs_string_kw("writeIntLE", 0, 0),
+        njs_string_kw("writeInt16BE", 0, 0),
+        njs_string_kw("writeInt16LE", 0, 0),
+        njs_string_kw("writeInt32BE", 0, 0),
+        njs_string_kw("writeInt32LE", 0, 0),
+        njs_string_kw("writeInt8", 0, 0),
+        njs_string_kw("writeUIntBE", 0, 0),
+        njs_string_kw("writeUIntLE", 0, 0),
+        njs_string_kw("writeUInt16BE", 0, 0),
+        njs_string_kw("writeUInt16LE", 0, 0),
+        njs_string_kw("writeUInt32BE", 0, 0),
+        njs_string_kw("writeUInt32LE", 0, 0),
+        njs_string_kw("writeUInt8", 0, 0),
+        njs_string_kw("written", 0, 0),
+    }
+};
+
+
+njs_cnst_hash_chunk_t njs_cnst_hash_predefined = {
+    .descr = {
+        .hash_mask = NJS_CNST_N_HASH_MASK,
+        .elts_size = NJS_CNST_N_VALUES,
+        .elts_count = 0,
+        .elts_deleted_count = 0,
+    }
+};
+
+
+njs_lvlhsh_t njs_cnst_hash = {
+    .slot = &njs_cnst_hash_predefined.descr,
+};
+
+
+uint32_t  njs_cnst_atom_id = NJS_SYMBOL_KNOWN_MAX;
+
+
+static njs_int_t
+njs_cnst_hash_test(njs_lvlhsh_query_t *lhq, void *data)
+{
+    size_t             size;
+    u_char             *start;
+    njs_value_t        *name;
+
+    name = data;
+
+    size = name->string.data->length;
+
+    if (lhq->key.length != size) {
+        return NJS_DECLINED;
+    }
+    start = (u_char *)name->string.data->start;
+
+    if (memcmp(start, lhq->key.start, lhq->key.length) == 0) {
+        return NJS_OK;
+    }
+
+    return NJS_DECLINED;
+}
+
+
+/*
+ *  Here is used statically allocated hash with correct size.
+ *  This hash is only filled in, and later used as read only one.
+ *  So, alloc/free are never used here.
+ */
+
+static const njs_lvlhsh_proto_t  njs_cnst_hash_proto
+    njs_aligned(64) =
+{
+    0,
+    njs_cnst_hash_test,
+    0,
+    0,
+};
+
+
+njs_int_t
+njs_cnst_hash_init(njs_lvlhsh_t *hash)
+{
+    u_char              *s;
+    size_t              l;
```

Comment:
Style: it is better to use standard variable names: `s` -> `start`, `l` -> `len`, or `length` for consistency.